addons/irlight/functions/fnc_initItemContextMenu.sqf

```diff
@@ -25,23 +25,30 @@
             private _baseClass = getText (configFile >> "CfgWeapons" >> _item >> "baseWeapon");
             _item != _baseClass + _variant
         }, {
-            params ["", "", "_item", "", "_variant"];
+            params ["_unit", "", "_item", "_slot", "_variant"];
 
             private _baseClass = getText (configFile >> "CfgWeapons" >> _item >> "baseWeapon");
 
-            ACE_player removePrimaryWeaponItem _item;
-            ACE_player addPrimaryWeaponItem (_baseClass + _variant);
-            playSound "click";
-
-            if (_turnedOn) then {
-                // Force update of flashlight
-                ACE_player action ["GunLightOff", ACE_player];
-
-                {
-                    ACE_player action ["GunLightOn", ACE_player];
-                    ACE_player action ["IRLaserOn", ACE_player];
-                } call CBA_fnc_execNextFrame;
+            switch (_slot) do {
+                case "RIFLE_POINTER": {
+                    _unit removePrimaryWeaponItem _item;
+                    _unit addPrimaryWeaponItem (_baseClass + _variant);
+                };
+                case "LAUNCHER_POINTER": {
+                    _unit removeSecondaryWeaponItem _item;
+                    _unit addSecondaryWeaponItem (_baseClass + _variant);
+                };
+                case "PISTOL_POINTER": {
+                    _unit removeHandgunItem _item;
+                    _unit addHandgunItem (_baseClass + _variant);
+                };
+                case "BINOCULAR_POINTER": {
```

Comment:
Is "BINOCULAR_POINTER" a thing? Not listed in https://github.com/CBATeam/CBA_A3/blob/master/addons/ui/fnc_addItemContextMenuOption.sqf#L58

Reply:
I was a bit skeptical of its existence, as there are no slots for bino attachments in Vanilla, but never checked. I'll remove it then.